src/hooks/useUSDCPrice.ts

```diff
@@ -40,42 +36,17 @@ export default function useUSDCPrice(currency?: Currency): Price<Currency, Token
       )
     }
 
-    // handle weth/eth
-    if (wrapped.equals(weth)) {
-      if (usdcPair) {
-        const price = usdcPair.priceOf(weth)
-        return new Price(currency, USDC, price.denominator, price.numerator)
-      } else {
-        return undefined
-      }
-    }
-    // handle usdc
-    if (wrapped.equals(USDC)) {
-      return new Price(USDC, USDC, '1', '1')
+    // use v2 price if available, v3 as fallback
+    if (v2USDCTrade) {
+      const { numerator, denominator } = v2USDCTrade.route.midPrice
+      return new Price(currency, USDC, denominator, numerator)
+    } else if (v3USDCTrade.state === V3TradeState.VALID && v3USDCTrade.trade) {
+      const { numerator, denominator } = v3USDCTrade.trade.route.midPrice
+      return new Price(currency, USDC, denominator, numerator)
```

Comment:
can you not just return midPrice? currencies should be correct

Reply:
midPrice is Price<Currency, Currency> rather than Price<Currency, Token>. Probably makes sense to build a Price<Currency, Token> rather than force the type of midPrice?

Reply:
oh. we don't particularly care about the currency though (we don't use it anywhere)

Reply:
hm changing to price<currency,currency> breaks a lot of downstream types (e.g. fiatvalue in currencyinputpanel). might just leave as-is to avoid scope creep